tests/test_commands.py

```diff
@@ -125,3 +135,60 @@ def test_validation_failed_message(self):
         self.assertIn("user", output.getvalue())
         self.assertIn("783", output.getvalue())
         self.assertIn("does not exist", output.getvalue())
+
+
+class ClearTokensTest(TestCase):
+    def setUp(self):
+        now = timezone.now()
+        earlier = now - timedelta(seconds=100)
+        later = now + timedelta(seconds=100)
+        UserModel = get_user_model()
+        user = UserModel.objects.create()
+
+        app = Application.objects.create(
+            client_type="confidential",
+            authorization_grant_type="client-credentials",
+            name="The App",
+        )
+        for i in range(100):
+            old = AccesstokenModel.objects.create(token="old access token {}".format(i), expires=earlier)
+            new = AccesstokenModel.objects.create(token="current access token {}".format(i), expires=later)
+            # make half of these Access Tokens have related Refresh Tokens, which prevent their deletion.
+            if i % 2:
+                RefreshtokenModel.objects.create(
+                    token="old refresh token {}".format(i),
+                    application=app,
+                    access_token=old,
+                    user=user,
+                )
+                RefreshtokenModel.objects.create(
+                    token="current refresh token {}".format(i),
+                    application=app,
+                    access_token=new,
+                    user=user,
+                )
+            GrantModel.objects.create(
+                user=user,
+                code="old grant code {}".format(i),
+                application=app,
+                expires=earlier,
+                redirect_uri="https://localhost/redirect",
+            )
+            GrantModel.objects.create(
+                user=user,
+                code="new grant code {}".format(i),
+                application=app,
+                expires=later,
+                redirect_uri="https://localhost/redirect",
+            )
```

Comment:
lint not running?

Reply:
It's what `black` appears to want. I always run `tox -e flake8` before committing. Also the pre-commit hooks do that as well I believe.